tap_googleads/custom_query_stream.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Any, Dict, List
+
+import humps
+import requests
+import sqlparse
+from singer_sdk.helpers._flattening import flatten_record
+
+from tap_googleads.streams import ReportsStream
+
+DATE_TYPES = ("segments.date", "segments.month", "segments.quarter", "segments.week")
+
+
+class CustomQueryStream(ReportsStream):
+    """Define custom stream."""
+
+    records_jsonpath = "$.results[*]"
+    primary_keys = []
+    replication_key = None
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Initialize the REST stream.
+
+        Args:
+            tap: Singer Tap this stream belongs to.
+        """
+        self.custom_query = kwargs.pop("custom_query")
+        self._query = self.custom_query["query"]
+        self.name = self.custom_query["name"]
+        super().__init__(*args, **kwargs)
+
+    @property
+    def gaql(self):
+        return self._query
+
+    @cached_property
+    def schema(self) -> dict:
+        """Return dictionary of record schema.
+
+        Dynamically detect the json schema for the stream.
+        This is evaluated prior to any records being retrieved.
+        """
+        local_json_schema = {
+            "type": "object",
+            "properties": {},
+            "additionalProperties": True,
+        }
```

Comment:
Can we use SDK typing here? Below suggestions all related.

```suggestion
        properties = th.PropertiesList(additional_properties=True)
```

Reply:
I don't actually know if `additional_properties=True` is the default (not clear from the SDK code), so I guess it's better to be explicit here.